### ISSUE TITLE
rent

### ISSUE THREAD

**Issue by rob-solana** [solana-labs/solana#4372]
#### Problem
 account state is currently rent-free, which isn't tenable for mainnet launch

 #### Proposed Solution
define and implement economics of rent, examples:

- target memory consumption+rent adjusted per epoch for target
- minimum balance per byte rule and accounts with state lose the state when balance drops below the per-byte minimum?
- Ethereum's proposals: https://www.reddit.com/r/ethereum/comments/ab1n2e/eli5_storage_rent/

 
 #### Complications
* vote accounts, storage accounts currently carry near-zero balance, but have non-near zero state
* where does rent go?
* when is rent collected? (epoch boundaries are maybe natural, but there are other large-ish operations in that spot)

**Comment by garious** [comment]
@rob-solana How far along is this? Ready for a mention in the v0.17.0 release notes?

**Comment by garious** [comment]
@rob-solana, ping

**Comment by rob-solana** [comment]
this completely missed v0.17

**Comment by ParthDesai** [comment]
We have implemented rent eco-system.

Rent is deducted per slot for read-write account as per configuration specified in genesis-config. Rent collected is distributed at the end of the slot, as per their stake-weight to validators after burning configurable amount of it.
